code/scripting/api/objs/subsystem.cpp

```diff
@@ -434,12 +434,18 @@ ADE_VIRTVAR(Target, l_Subsystem, "object", "Object targeted by this subsystem. I
 
 	ship_subsys *ss = sso->ss;
 
-	if(ADE_SETTING_VAR && objh && objh->IsValid())
-	{
-		ss->turret_enemy_objnum = OBJ_INDEX(objh->objp);
-		ss->turret_enemy_sig = objh->sig;
-		ss->targeted_subsys = NULL;
-		ss->scripting_target_override = true;
+	if(ADE_SETTING_VAR)	{
+		if (objh && objh->IsValid()) {
+			ss->turret_enemy_objnum = OBJ_INDEX(objh->objp);
+			ss->turret_enemy_sig = objh->sig;
+			ss->targeted_subsys = nullptr;
+			ss->scripting_target_override = true;
+		} else if (lua_isnil(L, 2)) {
+			ss->turret_enemy_objnum = -1;
+			ss->turret_enemy_sig = -1;
+			ss->targeted_subsys = nullptr;
+			ss->scripting_target_override = true;
+		}
```

Comment:
Should you decide to throw a lua error for the above issue, this would need to as well